### PR TITLE
Fixes for @skip directive

### PR DESCRIPTION
**Pull request by gilgardosh**
Related: #3882 
#5453 
#5326 
#5632

### CODE REVIEW

File: packages/plugins/typescript/operations/tests/ts-documents.spec.ts [hunk 1]
```diff
@@ -4870,6 +4875,7 @@ function test(q: GetEntityBrandDataQuery): void {
           id: String!
           name: String!
           address: Address!
+          firends: [User!]!
```
Comment (jorrit) [inline packages/plugins/typescript/operations/tests/ts-documents.spec.ts:4878]: typo

File: packages/plugins/typescript/operations/tests/ts-documents.spec.ts [hunk 2]
```diff
@@ -4884,6 +4890,9 @@ function test(q: GetEntityBrandDataQuery): void {
             address @include(if: $showAddress) {
               city
             }
+            firends @include(if: $isFriendly) {
```
Comment (jorrit) [inline packages/plugins/typescript/operations/tests/ts-documents.spec.ts:4893]: typo

File: packages/plugins/typescript/operations/tests/ts-documents.spec.ts [hunk 3]
```diff
@@ -4912,7 +4921,10 @@ function test(q: GetEntityBrandDataQuery): void {
           & { address?: Maybe<(
             { __typename?: 'Address' }
             & Pick<Address, 'city'>
-          )> }
+          )>, firends?: Maybe<Array<(
```
Comment (jorrit) [inline packages/plugins/typescript/operations/tests/ts-documents.spec.ts:4924]: typo

File: packages/utils/plugins-helpers/src/utils.ts [hunk 4]
```diff
@@ -30,3 +30,11 @@ export function getBaseType(type: GraphQLOutputType): GraphQLNamedType {
     return type;
   }
 }
+
+export function removeNonNullWrapper(type: GraphQLOutputType): GraphQLNamedType {
+  if (isNonNullType(type)) {
+    return type.ofType;
+  } else {
```
Comment (jorrit) [inline packages/utils/plugins-helpers/src/utils.ts:37]: This could be just a ternary, or the else can be removed.

**Comment by fnnzzz** [comment]
🙏

**Pull request merged**